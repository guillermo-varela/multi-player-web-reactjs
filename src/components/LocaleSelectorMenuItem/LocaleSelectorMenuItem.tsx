import React from 'react';
import NavDropdown from 'react-bootstrap/NavDropdown';
import { useDispatch } from 'react-redux';
import { useIntl } from 'react-intl';
import useTypedSelector from '../../hooks/useTypedSelector';
import { changeLocaleAction } from '../../actions/localeActions';

const LocaleSelectorMenuItem: React.FC = () => {
  const locale = useTypedSelector(state => state.locale);
  const dispatch = useDispatch();
  const intl = useIntl();

  return (
    <NavDropdown id='dropdownLocale' className='menu-link'
      title={`${intl.formatMessage({ id: 'root.locale.label' })}: ${locale}`} >
      <NavDropdown.Item as='button' onClick={() => dispatch(changeLocaleAction('en'))}>
        English
      </NavDropdown.Item>
      <NavDropdown.Item as='button' onClick={() => dispatch(changeLocaleAction('es'))}>
        Espa&ntilde;ol
      </NavDropdown.Item>
    </NavDropdown>
  );
};

export default LocaleSelectorMenuItem;
