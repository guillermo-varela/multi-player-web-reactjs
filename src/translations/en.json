{
  "component.about.code": "The source code for this project can be found at:",
  "component.about.codeSplit": "Route based code splitting",
  "component.about.csrf": "Use of custom HTTP headers in stateless/sessionless environments to prevent CSRF attacks",
  "component.about.forms.formik": "Forms in React using Formik",
  "component.about.hello": "Hi again :)",
  "component.about.internationalization": "Support for multiple languages (internationalization) using react-intl",
  "component.about.introduction": "This web application was developed to learn about:",
  "component.about.jwt": "Stateless/sessionless authentication on a server using \"Javascript Object Signing and Encryption\" (JOSE) and \"JSON Web Token\" (JWT)\"",
  "component.about.openAPI": "OpenAPI 3",
  "component.about.pixabay": "The multimedia content was fetched using the {link}.",
  "component.about.pixabay.name": "Pixabay API",
  "component.about.redux.typescript": "Redux using TypeScript",
  "component.about.rest.webflux": "REST services with reactive architecture (non-blocking) with Spring WebFlux and Spring Security",
  "component.error.message": "Ops! An error occurred...",
  "component.signin.button.sending": "Sending...",
  "component.signin.button.submit": "Sign In",
  "component.signin.credentials.hint": "Use user@test.com as email and password to test",
  "component.signin.empty.email": "Enter an email",
  "component.signin.incorrect.credentials": "Incorrect credentials",
  "component.signin.invalid.email": "Enter a valid email",
  "component.signin.invalid.password": "Enter a password",
  "component.signin.label.email": "Email",
  "component.signin.label.password": "Password",
  "menu.about": "About",
  "menu.home": "Home",
  "menu.images": "Images",
  "menu.more": "View More",
  "menu.sections": "Sections",
  "menu.signIn": "Sign In",
  "menu.signOut": "Sign Out",
  "menu.signOut.error.message": "Sign out failed",
  "menu.signOut.error.title": "Ops!",
  "menu.videos": "Videos",
  "multimedia.source": "Content from ",
  "multimedia.type.image": "Image",
  "multimedia.type.video": "Video",
  "nature.hello": "Hello!",
  "nature.introduction": "On this site you can find multimedia content about nature.",
  "nature.multimedia.endOfContent": "That's all for now...",
  "nature.title": "Nature",
  "root.loading": "Loading...",
  "root.locale.label": "Language",
  "root.nature.description": "Images and videos of flora and fauna.",
  "root.space.description": "Images and videos of the universe.",
  "space.hello": "Hello!",
  "space.images.introduction": "Multiple images related to outer space.",
  "space.introduction": "On this site you will find multimedia content about outer space.",
  "space.menu.title": "Space",
  "space.title": "Outer Space",
  "space.videos.introduction": "Mind-blowing videos from outer space."
}
