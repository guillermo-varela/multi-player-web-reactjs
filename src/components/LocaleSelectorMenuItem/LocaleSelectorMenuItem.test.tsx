import React from 'react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { IntlProvider } from 'react-intl';
import configureMockStore, { MockStoreEnhanced } from 'redux-mock-store';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import en from '../../translations/en.json';
import LocaleSelectorMenuItem from './LocaleSelectorMenuItem';

describe('<LocaleSelectorMenuItem />', () => {
  const mockStore = configureMockStore([thunk]);
  let store: MockStoreEnhanced<unknown, {}>;

  it('renders <NavDropdown />', () => {
    const wrapper = mount(buildComponent());
    expect(wrapper.find('NavDropdown')).toHaveLength(1);
  });

  it('matches snapshot', () => {
    const tree = renderer.create(buildComponent()).toJSON();
    expect(tree).toMatchSnapshot();
  });

  const buildComponent = (): JSX.Element => {
    store = mockStore({ locale: 'en' });
    return (
      <Provider store={store}>
        <IntlProvider locale={'en'} messages={en}>
          <LocaleSelectorMenuItem />
        </IntlProvider>
      </Provider>
    );
  }
});

