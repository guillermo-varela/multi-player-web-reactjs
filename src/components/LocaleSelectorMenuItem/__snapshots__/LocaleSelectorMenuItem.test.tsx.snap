// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocaleSelectorMenuItem /> matches snapshot 1`] = `
<div
  className="menu-link dropdown nav-item"
  onKeyDown={[Function]}
>
  <a
    aria-expanded={false}
    aria-haspopup={true}
    className="dropdown-toggle nav-link"
    href="#"
    id="dropdownLocale"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
  >
    Language: en
  </a>
</div>
`;
